src/pydisagg/ihme/sex_splitter.py

```diff
@@ -0,0 +1,234 @@
+
+    def parse_data(self, data: DataFrame) -> DataFrame:
+        name = "data"
+        validate_columns(data, self.data.columns, name)
+        data = data[self.data.columns].copy()
+        validate_index(data, self.data.index, name)
+        validate_nonan(data, name)
+        validate_positive(data, [self.data.val_sd], name)
+        return data
+
+    def parse_pattern(self, data: DataFrame, pattern: DataFrame) -> DataFrame:
+        name = "pattern"
+
+        if not all(
+            col in pattern.columns
+            for col in [self.pattern.val, self.pattern.val_sd]
+        ):
+            if not self.pattern.draws:
+                raise ValueError(
+                    "Must provide draws for pattern if pattern.val and "
+                    "pattern.val_sd are not available."
+                )
+            validate_columns(pattern, self.pattern.draws, name)
+            pattern[self.pattern.val] = pattern[self.pattern.draws].mean(axis=1)
+            pattern[self.pattern.val_sd] = pattern[self.pattern.draws].std(
+                axis=1
+            )
+
+        validate_columns(pattern, self.pattern.columns, name)
+        pattern = pattern[self.pattern.columns].copy()
+        validate_index(pattern, self.pattern.index, name)
+        validate_nonan(pattern, name)
+        validate_positive(pattern, [self.pattern.val_sd], name)
+        data_with_pattern = self._merge_with_pattern(data, pattern)
+        return data_with_pattern
+
+    def get_population_by_sex(self, population, sex_value):
+        return population[population[self.population.sex] == sex_value][
+            self.population.index + [self.population.val]
+        ].copy()
+
+    def parse_population(
+        self, data: DataFrame, population: DataFrame
+    ) -> DataFrame:
+        name = "population"
+        validate_columns(population, self.population.columns, name)
+
+        male_population = self.get_population_by_sex(
+            population, self.population.sex_m
+        )
+        female_population = self.get_population_by_sex(
+            population, self.population.sex_f
+        )
+
+        male_population.rename(
+            columns={self.population.val: "m_pop"}, inplace=True
+        )
+        female_population.rename(
+            columns={self.population.val: "f_pop"}, inplace=True
+        )
+
+        data_with_population = self._merge_with_population(
+            data, male_population, "m_pop"
+        )
+        data_with_population = self._merge_with_population(
+            data_with_population, female_population, "f_pop"
+        )
+
+        validate_nonan(data_with_population, name)
+        return data_with_population
+
+    def _merge_with_population(
+        self, data: DataFrame, population: DataFrame, pop_col: str
+    ) -> DataFrame:
+        data_with_population = data.merge(
+            population,
+            on=self.population.index,
+            how="left",
+        )
+        return data_with_population
```

Comment:
pop_col is never used after being passed through, and the merge is not actually happening, it is creating nans in the merge.